crates/octez/src/async/rollup.rs

```diff
@@ -0,0 +1,228 @@
+    /// HTTP endpoint of the rollup node RPC interface,
+    /// if None, use localhost with random port
+    pub rpc_endpoint: Option<Endpoint>,
+}
+
+impl OctezRollupConfigBuilder {
+    pub fn new(
+        octez_node_endpoint: Endpoint,
+        octez_client_base_dir: PathBuf,
+        data_dir: RollupDataDir,
+        address: SmartRollupHash,
+        operator: String,
+        boot_sector_file: PathBuf,
+    ) -> Self {
+        OctezRollupConfigBuilder {
+            binary_path: None,
+            data_dir,
+            octez_node_endpoint,
+            octez_client_base_dir,
+            address,
+            operator,
+            boot_sector_file,
+            rpc_endpoint: None,
+        }
+    }
+
+    pub fn set_binary_path(mut self, binary_path: &str) -> Self {
+        self.binary_path = Some(PathBuf::from(binary_path));
+        self
+    }
+
+    pub fn set_rpc_endpoint(mut self, rpc_endpoint: &Endpoint) -> Self {
+        self.rpc_endpoint = Some(rpc_endpoint.clone());
+        self
+    }
+
+    pub fn build(self) -> Result<OctezRollupConfig> {
+        Ok(OctezRollupConfig {
+            binary_path: self
+                .binary_path
+                .unwrap_or(PathBuf::from(DEFAULT_BINARY_PATH)),
+            octez_client_base_dir: self.octez_client_base_dir,
+            octez_node_endpoint: self.octez_node_endpoint,
+            data_dir: self.data_dir,
+            address: self.address,
+            operator: self.operator,
+            boot_sector_file: self.boot_sector_file,
+            rpc_endpoint: self.rpc_endpoint.unwrap_or_else(|| {
+                let uri = Uri::from_str(&format!("127.0.0.1:{}", unused_port())).unwrap();
+                Endpoint::try_from(uri).unwrap()
+            }),
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct OctezRollupConfig {
+    pub binary_path: PathBuf,
+    pub octez_client_base_dir: PathBuf,
+    pub octez_node_endpoint: Endpoint,
+    pub data_dir: RollupDataDir,
+    pub address: SmartRollupHash,
+    pub operator: String,
+    pub boot_sector_file: PathBuf,
+    pub rpc_endpoint: Endpoint,
+}
+
+pub struct OctezRollup {
+    binary_path: PathBuf,
+    /// Path to the directory where the rollup state & kernel preimages are stored
+    data_dir: PathBuf,
+    octez_client_base_dir: PathBuf,
+    octez_node_endpoint: Endpoint,
+    rpc_endpoint: Endpoint,
+}
+
+impl OctezRollup {
+    pub fn new(
+        binary_path: &Path,
+        data_dir: &Path,
+        octez_client_base_dir: &Path,
+        octez_node_endpoint: &Endpoint,
+        rpc_endpoint: &Endpoint,
+    ) -> Self {
+        Self {
+            binary_path: binary_path.to_path_buf(),
+            data_dir: data_dir.to_path_buf(),
+            octez_client_base_dir: octez_client_base_dir.to_path_buf(),
+            octez_node_endpoint: octez_node_endpoint.clone(),
+            rpc_endpoint: rpc_endpoint.clone(),
+        }
+    }
+}
+
+impl OctezRollup {
+    fn command(&self) -> Command {
```

Comment:
Why not merge this into `run`?

Reply:
though it would be more readable if we separate the commands with the global options from the specific command like run. Also, we can reuse if we were to implement other commands like the config init.